### ISSUE TITLE
Rule does not work for objects inside array destructuring

### ISSUE THREAD

**Issue by benmandr** [isaquediasm/eslint-plugin-destructure-depth#6]
Let's say we have:
```
const [{ propertyA: { propertyBofA } }] = [{ propertyA: { propertyBofA: 'value' }}];
```

In this case, rule does not get triggered.
